File: C++/B/1915B-notquitelatinsquare.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int t;
    cin >> t;
    while (t--)
    {
        string first, second, third;
        cin >> first >> second >> third;
        if (first.find('?') != string::npos)
        {
            if (first.find('A') == string::npos)
                cout << "A\n";
            else if (first.find('B') == string::npos)
                cout << "B\n";
            else
                cout << "C\n";
        }
        else if (second.find('?') != string::npos)
        {
            if (second.find('A') == string::npos)
                cout << "A\n";
            else if (second.find('B') == string::npos)
                cout << "B\n";
            else
                cout << "C\n";
        }
        else
        {
            if (third.find('A') == string::npos)
                cout << "A\n";
            else if (third.find('B') == string::npos)
                cout << "B\n";
            else
                cout << "C\n";
        }
    }
}

File: Python/B/1915B-notquitelatinsquare.py
t = int(input())
for _ in range(t):
    first = input()
    second = input()
    third = input()
    if "?" in first:
        if "A" not in first:
            print("A")
        elif "B" not in first:
            print("B")
        else:
            print("C")
    elif "?" in second:
        if "A" not in second:
            print("A")
        elif "B" not in second:
            print("B")
        else:
            print("C")
    else:
        if "A" not in third:
            print("A")
        elif "B" not in third:
            print("B")
        else:
            print("C")
